### PR TITLE
Migrate to using the FDL API for app re-opens

### PR DESCRIPTION
**Pull request by dmandar**

### CODE REVIEW

File: Example/DynamicLinks/Tests/FIRDynamicLinksTest.m [hunk 1]
```diff
@@ -1032,6 +1034,12 @@ - (void)testValidCustomDomainNames {
     @"https://a.firebase.com/mypath/mylink",  // Short FDL starting https://a.firebase.com/mypath
     @"https://a.firebase.com/mypath?link=abcd&test=1",  // Long FDL starting with
                                                         // https://a.firebase.com/mypath
+    @"https://a.firebase.com/mypath/?link=https://www.google.com&test=1"  // Long FDL coming from
```
Comment (mh6o) [inline Example/DynamicLinks/Tests/FIRDynamicLinksTest.m]: It'd be super-confusing to have links possibly expressed with/without slash before the query param start "?". Then that could mean either:
1) Short FDL that starts with "https://a.firebase.com/" with suffix "mypath"
2) Or, long FDL that starts with "https://a.firebase.com/mypath" with params "link=..."

On server, there's always a prior slash before the "?". Look at DdlTest L317 (internal)

Comment (mh6o) [reply Example/DynamicLinks/Tests/FIRDynamicLinksTest.m]: Can you explain how each can happen?

Comment (dmandar) [reply Example/DynamicLinks/Tests/FIRDynamicLinksTest.m]: are developers allowed create a long FDL without the slash before the "?" ? If not, i can we can remove the long FDL example without the '/  'at 1035 above.

Comment (dmandar) [reply Example/DynamicLinks/Tests/FIRDynamicLinksTest.m]: I only kept the test samples with a slash '/?'

Comment (mh6o) [reply Example/DynamicLinks/Tests/FIRDynamicLinksTest.m]: I don't think this ever happen (https://a.firebase.com/mypath?link=abcd&test=1), else if we do, we will confused between:
1) short FDL with suffix "mypath" & overrides link=abcd
2) or, long FDL

But at the same time, we don't have short URL overrides, we don't require / before query param for long FDL (already an ambiguity problem)..

**Comment by mh6o** [comment]
LGTM

**Comment by dmandar** [comment]
Closed incorrectly. Reopening.

File: Example/DynamicLinks/Tests/FIRDynamicLinksTest.m [hunk 2]
```diff
@@ -1027,10 +1029,10 @@ - (void)testValidCustomDomainNames {
   NSArray<NSString *> *urlStrings = @[
     @"https://google.com/mylink",             // Short FDL starting with 'https://google.com'
     @"https://google.com/one",                // Short FDL starting with 'https://google.com'
-    @"https://google.com?link=abcd",          // Long FDL starting with  'https://google.com'
+    @"https://google.com?/link=abcd",          // Long FDL starting with  'https://google.com'
     @"https://google.com/one/mylink",         // Long FDL starting with  'https://google.com/one'
     @"https://a.firebase.com/mypath/mylink",  // Short FDL starting https://a.firebase.com/mypath
-    @"https://a.firebase.com/mypath?link=abcd&test=1",  // Long FDL starting with
+    @"https://a.firebase.com/mypath/?link=abcd&test=1",  // Long FDL starting with
```
Comment (ryanwilson) [inline Example/DynamicLinks/Tests/FIRDynamicLinksTest.m]: Do they require a `/` now?

Comment (dmandar) [reply Example/DynamicLinks/Tests/FIRDynamicLinksTest.m]: Yes, based on Michael's comment.

**Pull request merged**